apps/artwork_purchase/client/purchase_form.coffee

```diff
@@ -6,45 +6,39 @@ User = require '../../../models/user.coffee'
 CurrentUser = require '../../../models/current_user.coffee'
 ArtworkInquiry = require '../../../models/artwork_inquiry.coffee'
 { formatMessage } = require '../helpers.coffee'
+UserFairAction = require '../../../models/user_fair_action.coffee'
 
 module.exports = class PurchaseForm extends Backbone.View
   _.extend @prototype, Form
 
-  initialize: ({ @artwork, @user }) ->
+  initialize: ({ @artwork }) ->
     @inquiry = new ArtworkInquiry
 
   submit: ({ success, error })->
     formData = @serializeForm()
-    message = formatMessage _.extend { @artwork, @user }, formData
+    message = formatMessage _.extend { @artwork }, formData
     { name } = formData
     @inquiry.set {
       name,
       message,
       artwork: @artwork.id,
       contact_gallery: true,
       inquiry_url: window.location.href,
-      user: @user
     }
-    @user.related().collectorProfile.findOrCreate() .then =>
 
     promises = [Q.promise (resolve) =>
       @inquiry.save null,
-        success: resolve
+        success: (model, response) -> resolve response
         error: (model, response, options) =>
           @$('.js-ap-form-errors').html @errorMessage(response)
           error?()
     ]
 
+    action = new UserFairAction
     if formData.attending
-      @user.related()
-        .collectorProfile.related()
-        .userFairActions
-        .attendFair @artwork.fair
+      promises.push Q.promise (resolve) ->
+        action.save fair_id: 'miami-project-2016',
```

Comment:
Backbone persistence actions like `.save` and `.fetch` actually return promises, so I don't _think_ you should need to wrap this.